src/OrchardCore/OrchardCore.Users.Core/Authentication/CacheTicketStore.cs

```diff
@@ -0,0 +1,155 @@
+using System;
+using System.IO;
+using System.Runtime.Serialization;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace OrchardCore.Users.Authentication;
+
+public class CacheTicketStore : ITicketStore
+{
+    private const string _keyPrefix = "ocauth-ticket";
+    private readonly IHttpContextAccessor _accessor;
+
+    public CacheTicketStore(IHttpContextAccessor accessor)
+    {
+        _accessor = accessor;
+    }
+
+    public async Task RemoveAsync(string key)
+    {
+        var cache = _accessor.HttpContext.RequestServices.GetService<IDistributedCache>();
+        var bytes = await cache.GetAsync($"{_keyPrefix}-{key}");
+        if (bytes != null && bytes.Length > 0)
+        {
+            await cache.RemoveAsync($"{_keyPrefix}-{key}");
```

Comment:
Can you confirm the previous `GetAsync` is required? I think just calling `RemoveAsync` in all cases should work